services/fluxmonitor.go

```diff
@@ -79,41 +78,46 @@ type FluxMonitor struct {
 	chTickerClose chan struct{}
 }
 
-func NewFluxMonitor(config FluxMonitorConfig, triggerJobRun chan subscriber.Event, sub store.Subscription) (*FluxMonitor, error) {
+func NewFluxMonitor(config FluxMonitorConfig, triggerJobRun chan subscriber.Event, blockchainManager blockchain.Manager) (*FluxMonitor, error) {
```

Comment:
Adding the BM directly as dependency, so FM doesn't care about subscriptions etc. Also mock BMs can be passed for testing as well. What do you think @boxhock?

Reply:
Yeah I like that!